course_catalog/etl/podcast.py

```diff
@@ -0,0 +1,195 @@
+from django.conf import settings
+import github
+import yaml
+from bs4 import BeautifulSoup as bs
+import requests
+from course_catalog.etl.utils import generate_unique_id
+from course_catalog.constants import PlatformType
+
+CONFIG_FILE_REPO = "mitodl/open-podcast-data"
+CONFIG_FILE_FOLDER = "podcasts"
+
+log = logging.getLogger()
+
+
+def github_podcast_config_files():
+    """
+    Function that returns a list of pyGithub files with youtube config channel data
+
+    Returns:
+        A list of pyGithub contentFile objects
+    """
+    github_client = github.Github()
+    repo = github_client.get_repo(CONFIG_FILE_REPO)
+
+    return repo.get_contents(CONFIG_FILE_FOLDER, ref=settings.OPEN_PODCAST_DATA_BRANCH)
+
+
+def validate_podcast_config(podcast_config):
+    """
+    Validates a playlist config
+
+    Args:
+        podcast_config (dict): the podcast config object
+
+    Returns:
+        list of str:
+            list of errors or an empty list if no errors
+    """
+    errors = []
+
+    if not podcast_config:
+        errors.append("podcast config data is empty")
+        return errors
+
+    if not isinstance(podcast_config, dict):
+        errors.append("Podcast data should be a dict")
+        return errors
+
+    for required_key in ["rss_url", "offered_by", "website"]:
+        if required_key not in podcast_config:
+            errors.append(f"Required key '{required_key}' is not present")
+
+    return errors
+
+
+def get_podcast_configs():
+    """
+    Fetch podcast configs from github
+
+    Returns:
+        list of dict:
+            a list of configuration objects
+    """
+    podcast_configs = []
+
+    for file in github_podcast_config_files():
+        try:
+            podcast_config = yaml.safe_load(file.decoded_content)
+
+            errors = validate_podcast_config(podcast_config)
+
+            if errors:
+                log.error(
+                    "Invalid youtube podcast config for path=%s errors=%s",
+                    file.path,
+                    errors,
+                )
+
+            podcast_configs.append(podcast_config)
+        except yaml.scanner.ScannerError:
+            log.exception("Error parsing podcast config for path=%s", file.path)
+            continue
+
+    return podcast_configs
+
+
+def extract():
+    """
+    Function for extracting podcast data
+
+    Returns:
+        A generator
+    """
+    configs = get_podcast_configs()
+
+    if not configs:
+        return
+
+    for playlist_config in configs:
+        rss_url = playlist_config["rss_url"]
+        try:
+            response = requests.get(rss_url)
+        except ConnectionError:
+            log.error("Invalid rss url %s", rss_url)
+
+        feed = bs(response.content, "xml")
+        yield (feed, playlist_config)
+
+
+def transform_episode(rss_data, offered_by, topics):
+    """
+    Transform a podcast episode into our normalized data
+
+    Args:
+        episode_rss (tuple): the extracted episode data
```

Comment:
Should be `rss_data`, and it looks like a different type than a tuple from the usage in the code below